src/display_control.rs

```diff
@@ -47,6 +47,29 @@ fn are_display_names_unique(displays: &[Display]) -> bool {
     displays.iter().all(|display| hash.insert(display_name(display, None)))
 }
 
+fn try_switch_display(handle: &mut Handle, display_name: &str, input: InputSource) {
+	match handle.get_vcp_feature(INPUT_SELECT) {
+		Ok(raw_source) => {
+			if raw_source.value() == input.value() {
```

Comment:
A quick hack that makes this work for me:
```suggestion
			if raw_source.value() & 0xff == input.value() {
```

Reply:
This is weird... Only the lowest byte is _supposed_ to have a value, all the rest are supposed to be zeros:

<img width="745" alt="image" src="https://user-images.githubusercontent.com/625874/170888100-460e907f-51df-4e15-8cc1-28625f952f8a.png">

I think this `& 0xff` is safe here, since all the input values should fit within a byte.

Reply:
Definitely the monitor's fault, `sh` is always supposed to be `0x00`.

ddc-rs always uses two bytes when decoding the command:

https://github.com/arcnmx/ddc-rs/blob/2f6c686a487cc59cdae4202853c59febfd733c87/src/commands.rs#L109-L115

So masking here is probably the appropriate thing to do.  Perhaps a slightly cleaner way would be to just read `raw_source.sl`.